MapleServer2/Types/Skill.cs

```diff
@@ -0,0 +1,21 @@
+namespace MapleServer2.Types {
+    public struct Skill {
+        public int Id { get; private set; }
+        public short Level { get; private set; }
+        public byte Learned { get; private set; }
+        public string Feature { get; private set; }
+        public int[] Sub { get; private set; }
+
+        public static Skill skill(int id, short level, byte learned, string feature = "", int[] sub = null) {
```

Comment:
Does this method need to exist if it's just calling the constructor and passing in everything 1 to 1? Can't the constructor just be called directly instead?

Reply:
To add on to the previous comment, if done this way, these objects should be cached somewhere because that's the general usage of static as some sort of factory. It's not immediately obvious how this is used but from the example in how a new character is made and how job skills are loaded later, it looks it this is a factory of some sort so it should be cached.

Reply:
Tbh I copied the structure and then changed it from another type in Maple2.Data iirc and yeah it can just use the constructor.